SparkPost/2web2ui
# SA-1665 - (UI) Comparators - Filters update

[SA-1665](https://sparkpost.atlassian.net/browse/SA-1665)

### What Changed

- Adds new `<AddFiltersSectionV2 />` component that renders behind the `"allow_report_filters_v2"` account UI flag - **should this be a user UI flag?**
- Adds new `<Uppercase />` and `<Emphasized />` re-usable text components
- Adds new `<RadioButtonGroup />` and `<RadioButtonGroup.Button />` components
- Adds new `useMultiEntry` hook - **should this be named** `useComboBoxTextField`**?**

### How To Test

- Go to `/signals/analytics` with the `"allow_report_filters_v2"` account flag set to `true`
- Click on "Add Filters"
- Verify the form behaves according to the mocks
- Verify submitted form data is reformatted in to the expected API structure

### To Do

- [x] Update handling of removing a filter according to: https://msys.slack.com/archives/G0195L0U5LH/p1601921523000600?thread_ts=1601664409.004000&cid=G0195L0U5LH
- [x] Focus management isn't working quite right - *could reviewers help me troubleshoot a bit?*
- [x] Add Cypress tests
- [x] Handle form submission/query param stuff on submit - *but this in Patrick's ticket handling JSON parsing?*
- [x] The "AND / OR" radio buttons should always render if _any_ filter has values within a grouping
- [x] Verify subaccounts search based on criteria from message w/ Greg:

> If a subaccount makes a request using the query_string parameter, the query will always be restricted to data for that subaccount. e.g. if subaccount 10  makes a request for subaccounts eq 12, 14, 16 , they will not in fact get data for subaccounts 12, 14, 16, only subaccount 10

- [x] Ensure subaccounts can only be compared by `"eq"` and `"not.eq"`
- [x] Fix re-mapping function on form submission
- [x] Overflow hidden vertically when the form gets too long
- [x] Fix initial selection of the filter "Type" field - placeholder is rendering funny
- [x] Work with API team on restructuring endpoint
- [x] Adding a grouping causes a rendering error
- [x] Conditionally rendering the "AND +" button between groupings
- [x] Add "AND/OR" switch and conditional "AND +" and "OR +" buttons
- [x] Break up `<AddFiltersSectionV2 />` in to configuration variables and components outside of the main file
- [x] Get initial state rendering
- [x] Add helpers to restructure data that is more React-friendly
- [x] Prepare form fields
- [x] Prepare layout according to mocks
- [x] Add `useMultiEntry` hook
